datasets/flwr_datasets/partitioner/partitioner.py

```diff
@@ -30,6 +31,10 @@ class Partitioner(ABC):
 
     def __init__(self) -> None:
         self._dataset: Optional[Dataset] = None
+        event(
+            EventType.PARTITIONER_CREATED,
+            {"class_name": self.__class__.__name__, "id": id(self)},
+        )
```

Comment:
This should move out of the constructor. Perhaps call it lazily in `load_partition` and other relevant methods?